docs/architecture/tablets.rst

```diff
@@ -0,0 +1,124 @@
+=========================================
+Data Distribution with Tablets
+=========================================
+
+A ScyllaDB cluster is a group of interconnected nodes. The data of the entire 
+cluster has to be distributed as evenly as possible across those nodes.
+
+ScyllaDB is designed to ensure a balanced distribution of data by storing data
+in tablets. When you add or remove nodes to scale your cluster, add or remove
+a datacenter, or replace a node, tablets are moved between the nodes to keep
+the same number on each node. In addition, tablets are balanced across shards
+in each node.
+
+This article explains the concept of tablets and how they let you scale your
+cluster quickly and seamlessly.
+
+Data Distribution
+-------------------
+
+ScyllaDB distributes data by splitting tables into tablets. Each tablet has 
+its replicas on different nodes, depending on the RF (replication factor). Each
+partition of a table is mapped to a single tablet in a deterministic way. When you
+query or update the data, ScyllaDB can quickly identify the tablet that stores
+the relevant partition. 
+
+The following example shows a 3-node cluster with a replication factor (RF) of
+3. The data is stored in a table (Table 1) with two rows. Both rows are mapped
+to one tablet (T1) with replicas on all three nodes.
+
+.. image:: images/tablets-cluster.png
+
+.. TODO - Add a section about tablet splitting when there are more triggers,
+   like throughput. In 6.0, tablets only split when reaching a threshold size
+   (the threshold is based on the average tablet data size).
+
+Load Balancing
+==================
+
+ScyllaDB autonomously moves tablets to balance the load. This process
+is managed by a load balancer mechanism and happens independently of
+the administrator or node operations such as bootstrap or decommission.
+The tablet load balancer decides where to migrate the tablets, either within
+the same node to balance the shards or across the nodes to balance the global
+load in the cluster.
+
+As a table grows, each tablet can split into two, creating a new tablet.
+The load balancer can migrate the split halves independently to different nodes
+or shards.
+
+The load-balancing process takes place in the background and is performed
+without any service interruption.
+
+Scaling Out
+=============
+
+A tablet can be dynamically migrated to either an existing node or a new node – on
+bootstrap. Paired with consistent topology updates with Raft, tablets allow
+you to add multiple nodes simultaneously. Existing nodes stream data to the new
+ones, and the system load eventually converges to an even distribution as 
+the process completes. 
+
+With tablets, streaming data to a new node is fast and efficient, because:
+
+* Entire SStables are streamed, so less data is read at the source.
```

Comment:
> Entire SStables are streamed

If this is about file-based streaming, then this is enterprise-only feature. The advantage is not that we read less, we read the same amount, but we stream less over network (stable files are more dense) and consume less CPU because we avoid the cost of deserialization of sstable files and reserialization into mutation fragment stream.

In open-source we stream mutation fragments like with vnodes. We may avoid index lookups at vnode boundaries, but the saving of disk reads from that is marginal if data dominates.

Reply:
good point, it is probably more of an issue only with small partition where reading the index consumes similar bandwidth to reading the data itself.